File: frontends/configurator/www/model/config/entities/rule_definition_action_parameter.js
import {Entities} from '../entities.js';
import {EntitiesHooks} from '../entities_hooks.js';
import {Node} from '../node.js';
import {RuleEntities} from '../rule_entities.js';

export class RuleDefinitionActionParameter extends Node {
	static getProperties() {
		return {
			id: {type: 'string'},
			label: {type: 'string'},
			type: {type: 'string'},
			dataEntity: {type: 'string'},
			configurationEntity: {type: 'string'},
			options: {type: 'string'}
		};
	}

	constructor(values) {
		super();
		this.id = undefined;
		this.label = undefined;
		this.type = undefined;
		this.dataEntity = undefined;
		this.configurationEntity = undefined;
		this.options = undefined;
		EntitiesHooks?.CreateNode.call(this, values);
	}

	getConfigurationEntity() {
		return Entities[this.configurationEntity];
	}
	getDataEntity() {
		return RuleEntities[this.dataEntity];
	}
}

File: backend/src/main/java/ch/rodano/batch/pojo/RuleDefinitionActionParameter.java
package ch.rodano.batch.pojo;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class RuleDefinitionActionParameter {

	private String id;
	private String label;
	private String dataEntity;
	private String configurationEntity;
	private String options;

	public String getId() {
		return id;
	}

	public void setId(final String id) {
		this.id = id;
	}

	public String getLabel() {
		return label;
	}

	public void setLabel(final String label) {
		this.label = label;
	}

	public String getDataEntity() {
		return dataEntity;
	}

	public void setDataEntity(final String dataEntity) {
		this.dataEntity = dataEntity;
	}

	public String getConfigurationEntity() {
		return configurationEntity;
	}

	public void setConfigurationEntity(final String configurationEntity) {
		this.configurationEntity = configurationEntity;
	}

	public String getOptions() {
		return options;
	}

	public void setOptions(final String options) {
		this.options = options;
	}

	@Override
	public String toString() {
		return "RuleDefinitionActionParameter{" +
			"id='" + id + '\'' +
			", label='" + label + '\'' +
			", dataEntity='" + dataEntity + '\'' +
			", configurationEntity='" + configurationEntity + '\'' +
			", options='" + options + '\'' +
			'}';
	}
}
